x/gateway/keeper/msg_server_stake_gateway.go

```diff
@@ -31,17 +33,25 @@ func (k msgServer) StakeGateway(
 		gateway = k.createGateway(ctx, msg)
 		coinsToDelegate = *msg.Stake
 	} else {
-		logger.Info(fmt.Sprintf("Gateway found. Updating gateway stake for address %s", msg.Address))
+		logger.Info(fmt.Sprintf("Gateway found. About to try and update gateway for address %s", msg.Address))
 		currGatewayStake := *gateway.Stake
 		if err = k.updateGateway(ctx, &gateway, msg); err != nil {
+			logger.Error(fmt.Sprintf("could not update gateway for address %s due to error %v", msg.Address, err))
```

Comment:
I recommend using `%q` instead of `%s` to make an empty string clear in the resulting log line.

```suggestion
		logger.Info(fmt.Sprintf("Gateway found. About to try and update gateway for address %q", msg.Address))
		currGatewayStake := *gateway.Stake
		if err = k.updateGateway(ctx, &gateway, msg); err != nil {
			logger.Error(fmt.Sprintf("could not update gateway for address %q due to error %v", msg.Address, err))
```